servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/AlpnChannelSingle.java

```diff
@@ -94,20 +97,34 @@ protected void configurePipeline(final ChannelHandlerContext ctx, final String p
 
         @Override
         protected void handshakeFailure(final ChannelHandlerContext ctx, final Throwable cause) {
-            LOGGER.warn("{} TLS handshake failed:", ctx.channel(), cause);
+            LOGGER.debug("{} TLS handshake failed:", ctx.channel(), cause);
```

Comment:
what's the motivation behind dropping it down to debug from warn?

Reply:
I thought it's useless because on the client-side users always get an exception from the client. Thinking about it again, it's useful on the server-side. I will revert this change and see if I can extract the ALPN changes into a separate PR. It was a general problem that handshake status was not reported when ALPN is used 😞

Reply:
👍

Reply:
Extracted ALPN-related changes to #2712. For re-review, please ignore failure propagation changes in `AlpnChannelSingle`.